backend/business-partner-agent/src/main/java/org/hyperledger/bpa/controller/ProofTemplateController.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hyperledger.bpa.controller;
+
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.annotation.*;
+import io.micronaut.scheduling.TaskExecutors;
+import io.micronaut.scheduling.annotation.ExecuteOn;
+import io.micronaut.security.annotation.Secured;
+import io.micronaut.security.rules.SecurityRule;
+import io.micronaut.validation.Validated;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.hyperledger.bpa.controller.api.prooftemplates.ProofTemplate;
+import org.hyperledger.bpa.impl.ProofTemplateManager;
+import org.hyperledger.bpa.impl.verification.prooftemplates.ValidUUID;
+import org.hyperledger.bpa.model.BPAProofTemplate;
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Controller("/api/proof-templates")
+@Tag(name = "Proof Template Management")
+@Validated
+@Secured(SecurityRule.IS_AUTHENTICATED)
+@ExecuteOn(TaskExecutors.IO)
+public class ProofTemplateController {
+
+    @Inject
+    private ProofTemplateManager proofTemplateManager;
+
+    @Get
+    public HttpResponse<List<ProofTemplate>> listProofTemplates() {
+        return HttpResponse.ok(
+                proofTemplateManager.listProofTemplates()
+                        .map(BPAProofTemplate::toRepresentation)
+                        .collect(Collectors.toList()));
+    }
+
+    @Post
+    public HttpResponse<ProofTemplate> addProofTemplate(@Valid ProofTemplate template) {
```

Comment:
I have to check that in a test.

Reply:
Yes. Works fine without @Body

Reply:
It works in the unit test below because you are calling it directly. See AdminControllerTest for an http example. My guess is that if you call this method via http template will be null without `@Body`.

Reply:
I forgot to push that commit. https://github.com/hyperledger-labs/business-partner-agent/pull/539/commits/31572378dba8f4ef81f2b149826e9a99e879dc02

Reply:
And it turned out the test was green, but the front end was unable to call that endpoint successfully.